### PR TITLE
White-paper table generator

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
    Wrote parser

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/Generation.kt [hunk 1]
```diff
@@ -47,6 +65,37 @@ private fun generateWarningNames() {
...
+private fun generateAvailableRules() {
+    val ruleMap = File("info/rules-mapping.md").readLines().drop(2)
+            .map { it.drop(1).dropLast(1).split("|") }
+            .map { RuleDescription(it[0].replace("\\s+".toRegex(), ""), it[1], it[2], null) }
+            .map { it.ruleName to it }.toMap()
+    File("info/available-rules.md").readLines()
+            .drop(2)
+            .map { it.drop(1).dropLast(1).split("|") }
+            .map { it[2].replace("\\s+".toRegex(), "") to it[5] }
+            .forEach { ruleMap[it.first]?.config = it.second }
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/Generation.kt:78]: I think you should either log a warning if `it.first` is not present in a map, or throw an error, if it means docs are inconsistent with each other.

**Review by petertrr — CHANGES_REQUESTED** [review]

File: wp/sections/feature.tex [hunk 2]
```diff
@@ -3,18 +3,27 @@
 \subsection{Configuration file}
 \par
 It's worth starting with the configuration file. This is a file in which the user can manually turn rules on and off or configure the rules settings. Below is one of the rules in the configuration file.
-
```
Comment (petertrr) [inline wp/sections/feature.tex:6]: Is it okay that there are changes in many tex files? Are they part of this change?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/GenerationAvailableRules.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        .associateBy { it.ruleName }
+    File("info/available-rules.md").readLines()
+        .drop(2)
+        .map { it.drop(1).dropLast(1).split("|") }
+        .map { it[2].replace("\\s+".toRegex(), "") to it[5] }
+        .forEach { ruleMap[it.first]!!.config = it.second}
+    val newText = File("wp/sections/appendix.tex").readLines().toMutableList()
+    newText.removeAll(newText.subList(newText.indexOf("\\section*{available-rules}") + 1, newText.indexOf("\\section*{\\textbf{Diktat Coding Convention}}")))
+    var index = newText.indexOf("\\section*{available-rules}") + 1
+    AUTO_TABLE.trimIndent().split("\n").forEach { newText.add(index++, it) }
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/GenerationAvailableRules.kt:31]:
```suggestion
    AUTO_TABLE.trimIndent().lines().forEach { newText.add(index++, it) }
```

**Review by petertrr — CHANGES_REQUESTED** [review]
As I can see, your generator only changes `appendix.tex`, what are other changes in wp related files?

File: wp/sections/feature.tex (hunk 2 above)
Comment (kentr0w) [reply wp/sections/feature.tex:6]: This is because of merge or because we made minor changes to WP and I don't want to create a new PR for this.

**Review by petertrr — APPROVED** [review]
LGTM

**Pull request merged**